src/@next/components/organisms/CartRow/CartRow.tsx

```diff
@@ -40,28 +40,33 @@ export const CartRow: React.FC<IProps> = ({
   attributes = [],
   onRemove,
 }: IProps) => {
-  const [tempQuantity, setTempQuantity] = useState<number | string>(quantity);
+  const [tempQuantity, setTempQuantity] = useState<string>(quantity.toString());
   const [isTooMuch, setIsTooMuch] = useState(false);
 
   const handleBlurQuantityInput = () => {
-    const newQuantity =
-      typeof tempQuantity === "number"
-        ? tempQuantity
-        : parseInt(tempQuantity, 10);
-    const notEnoughQuantity = isNaN(newQuantity) || newQuantity <= 0;
-    const tooMuchQuantity = newQuantity > maxQuantity;
-    if (notEnoughQuantity && !tooMuchQuantity) {
-      setTempQuantity(quantity);
-    } else if (!notEnoughQuantity && tooMuchQuantity) {
-      setTempQuantity(maxQuantity);
-      onQuantityChange(maxQuantity);
+    let newQuantity = parseInt(tempQuantity, 10);
+    let newTempQuantity = newQuantity.toString(); // To make newTempQuantity consistent with newQuantity
+
+    if (isNaN(newQuantity) || newQuantity <= 0) {
+      newQuantity = quantity;
+      newTempQuantity = quantity.toString();
+    } else if (newQuantity > maxQuantity) {
+      newQuantity = maxQuantity;
+      newTempQuantity = maxQuantity.toString();
+    }
+
+    if (quantity !== newQuantity) {
+      onQuantityChange(newQuantity);
+    }
+    if (tempQuantity !== newTempQuantity) {
+      setTempQuantity(newTempQuantity);
     }
 
-    setIsTooMuch(false);
+    setIsTooMuch(false); // It was set to maxQuantity when newQuantity > maxQuantity, thus always false.
```

Comment:
Move this line to line 57 and comment will be unnecessary.

Reply:
Fixed